### DISCUSSION TITLE
Check for all my formulae and casks compatibility before upgrading MacOS

### DISCUSSION THREAD

**Discussion by superbonaci** [Homebrew/discussions#4811] [Everyday usage]
### Output of `brew config`

```shell
not applicable
```

### Output of `brew doctor`

```shell
not applicable
```

### Description of issue

Currently with MacOS 16.6 (Apple Silicon), and would like to upgrade to 14.0.
How can I check if all my formulae and casks will be compatible from command line with x version of MacOS before upgrading?
Something like:
```
brew checkcompat 14.0
```
or
```
brew chkos Sonoma
```

the output could be something:
```
Installed compatibility with: macOS 14.0 arm64
==> Formulae
md5deep         :yes
openjdk         :no

==> Casks
mucommander     :yes
```
Maybe this option already exists?

**Comment by SMillerDev** [comment]
Generally, formulae work on any version you installed them with or newer. Casks are impossible to check programmatically.

**Comment by superbonaci** [comment]
So basically I have to risk it...
